src/contexts/Dashboard.js

```diff
@@ -0,0 +1,222 @@
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import weekOfYear from "dayjs/plugin/weekOfYear";
+import PropTypes from "prop-types";
+import React, {
+  useCallback,
+  useEffect,
+  useReducer,
+  useState,
+  createContext,
+  useContext,
+  useMemo,
+} from "react";
+
+import { GLOBAL_CHART } from "../apollo/queries";
+import { useTimeframe } from "./Application";
+import { getTimeframe } from "../utils";
+import { clients } from "../apollo/client";
+import { SupportedNetwork } from "../constants";
+
+// format dayjs with the libraries needed
+dayjs.extend(utc);
+dayjs.extend(weekOfYear);
+
+const DashboardDataContext = createContext();
+
+export const useDashboardDataContext = () => {
+  return useContext(DashboardDataContext);
+};
+
+const SUPPORTED_CLIENTS = [
+  {
+    network: SupportedNetwork.MAINNET,
+    client: clients[SupportedNetwork.MAINNET],
+  },
+  { network: SupportedNetwork.XDAI, client: clients[SupportedNetwork.XDAI] },
+  {
+    network: SupportedNetwork.ARBITRUM_ONE,
+    client: clients[SupportedNetwork.ARBITRUM_ONE],
+  },
+];
+
+const INITIAL_STATE = { stackedChartData: {} };
+const UPDATE_CHART = "UPDATE_CHART";
+const RESET = "RESET";
+
+function reducer(state, { type, payload }) {
+  switch (type) {
+    case UPDATE_CHART: {
+      const { daily } = payload;
+      return {
+        ...state,
+        stackedChartData: {
+          daily,
+        },
+      };
+    }
+
+    case RESET: {
+      return INITIAL_STATE;
+    }
+
+    default: {
+      throw Error(
+        `Unexpected action type in DashboardContext reducer: '${type}'.`
+      );
+    }
+  }
+}
+
+export default function Provider({ children }) {
+  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
+  const updateChart = useCallback((daily) => {
+    dispatch({
+      type: UPDATE_CHART,
+      payload: {
+        daily,
+      },
+    });
+  }, []);
+
+  return (
+    <DashboardDataContext.Provider
+      value={useMemo(
+        () => [
+          state,
+          {
+            updateChart,
+          },
```

Comment:
```suggestion
            updateChart
```
You can put `updateChart` directly here no need for an object